In [1]:

from pathlib import Path
import numpy as np
import pandas as pd

DATA_PATH = Path('AB_NYC_2019.csv')
df = pd.read_csv(DATA_PATH)
df['price'] = pd.to_numeric(df['price'], errors='coerce')
df['reviews_per_month'] = df['reviews_per_month'].fillna(0)
df = df.dropna(subset=['price']).copy()
df['y_log'] = np.log1p(df['price'])
price_cap = df['price'].quantile(0.99)
df['price_for_viz'] = df['price'].clip(upper=price_cap)

price_median = df['price'].median()
ninety_ninth = price_cap
room_means = df.groupby('room_type')['price'].mean().sort_values(ascending=False)
neighbourhood_means = df.groupby('neighbourhood_group')['price'].mean().sort_values(ascending=False)
latitude_range = df['latitude'].min(), df['latitude'].max()
longitude_range = df['longitude'].min(), df['longitude'].max()
insights = {
    'median_price': float(price_median),
    'p99_price': float(ninety_ninth),
    'room_means': room_means.to_dict(),
    'neighbourhood_means': neighbourhood_means.to_dict(),
    'lat_range': latitude_range,
    'lon_range': longitude_range
}
insights


{'median_price': 106.0,
 'p99_price': 799.0,
 'room_means': {'Entire home/apt': 211.79424613325986,
  'Private room': 89.78097285675894,
  'Shared room': 70.12758620689655},
 'neighbourhood_means': {'Manhattan': 196.8758136743456,
  'Brooklyn': 124.38320732192598,
  'Staten Island': 114.81233243967829,
  'Queens': 99.51764913519237,
  'Bronx': 87.4967919340055},
 'lat_range': (np.float64(40.49979), np.float64(40.91306)),
 'lon_range': (np.float64(-74.24442), np.float64(-73.71299))}

**Note:** The code above turns the cleaned dataset into summary numbers—median, 99th percentile, room/borough means, and latitude/longitude ranges—so you can cite explicit figures in the commentary.



# Key early insights (Phase I)
- Nightly prices are right-skewed: the median is about $106 while the 99th percentile is roughly $799, so most listings are moderately priced but a few luxury units stretch the tail.
- Entire home/apartment listings average about $212 per night, more than double private rooms (~$90) and well above shared rooms (~$70), reinforcing room type as a strong price signal.
- Manhattan commands the highest mean price (~$197) followed by Brooklyn (~$124), whereas Bronx and Queens tend to cluster under $100, matching expectations from the box/point plots.
- Filling `reviews_per_month` with zeros preserves host activity signal for listings with no reviews yet, and dropping the rare missing prices avoids skew when summarizing prices.
- Mapping latitude/longitude (sampled for clarity) shows dense, high-price clusters in Manhattan and parts of Brooklyn, suggesting spatial coordinates can help distinguish premium areas.


**Tip:** Run this notebook to refresh the summary dictionary before exporting slides, then copy the bullets below straight into your deck.
